git-tool-providers/azure-dev-ops.mjs

```diff
@@ -1,33 +1,144 @@
-import * as azdev from "azure-devops-node-api";
-import * as dotenv from 'dotenv' 
-dotenv.config('../env');
+import * as azdev from 'azure-devops-node-api'
+import * as dotenv from 'dotenv'
+dotenv.config('../env')
+import { DateTime } from 'luxon'
+import { userMapping } from '../userMapping.mjs'
 
 const orgUrl = process.env.ORG_URL
 const token = process.env.AZURE_PERSONAL_ACCESS_TOKEN
-const repoId = process.env.REPO_ID
-const project = process.env.PROJECT_NAME
+const projects = process.env.PROJECT_NAMES.split(', ')
+const t = 'days between commits'
+const recent = `sum of all ${t}`
+const avgDays = `avg ${t}`
 
 export async function getAzureDevOpsProvider(orgUrl, token) {
-    let authHandler = azdev.getPersonalAccessTokenHandler(token); 
-    let webApi = new azdev.WebApi(orgUrl, authHandler);  
-    let gitApiObject = await webApi.getGitApi();
-    return gitApiObject;
+  let authHandler = azdev.getPersonalAccessTokenHandler(token)
+  let webApi = new azdev.WebApi(orgUrl, authHandler)
+  let gitApiObject = await webApi.getGitApi()
+
+  return gitApiObject
 }
 
 export async function getCommitersByName() {
-    let gitApiObject = await getAzureDevOpsProvider(orgUrl, token)
-    const commits = await gitApiObject.getCommits(repoId, {$skip: 0, $top: 15000}, project)
```

Comment:
we could also [request for all the commits for a single committer](https://github.com/microsoft/azure-devops-node-api/blob/e3daeea0d23b9d011a88b120e53eb4f8aa475542/api/interfaces/GitInterfaces.ts#L2133) by updating the [search criteria of the getCommits method](https://github.com/microsoft/azure-devops-node-api/blob/9ccac86e5bb38c553c433b4e300d920e7e3e2b40/api/GitApi.ts#L1227)

Reply:
alternatively we could also [request for all the commits for a single author](https://github.com/microsoft/azure-devops-node-api/blob/e3daeea0d23b9d011a88b120e53eb4f8aa475542/api/interfaces/GitInterfaces.ts#L2069) by updating the [search criteria of the getCommits method](https://github.com/microsoft/azure-devops-node-api/blob/9ccac86e5bb38c553c433b4e300d920e7e3e2b40/api/GitApi.ts#L1227)

Reply:
Because oh how git stores names doing query by name would be difficult, this is the same issue that caused us to have the userMapping.mjs file.